From 4c06080c61cd7338dabfabde05b2efb1ef5c37b7 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 08 Oct 2020 12:32:18 +0300
Subject: [PATCH] [SW-17195] preboot: fuse: add unimplemented string

On Goya1/2 and later SoCs, the eFuse version reading is still
unimplemented and prints "unknown" string which is misleading and
makes people think there is an eFuse reading problem.
Add "not implemented" string to the "weak" implementation, so it will
be clear this is not a problem of reading eFuse.

Change-Id: Ic8622924fc95bf1d38296b1d4c18d6aead7890b0
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/preboot.c b/preboot/preboot.c
index 82389ce..0e8221c 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -111,11 +111,23 @@
 
 #define FUSE_VER_PREFIX		"Fuse version "
 #define FUSE_VER_UNKNOWN	"unknown"
+#define FUSE_VER_NOT_IMPLEMENT	"not implemented"
+
 static char preboot_fuse_ver_str[VERSION_MAX_LEN] = FUSE_VER_PREFIX;
 
 __weak int hl_fuse_get_version(char *version, size_t size)
 {
-	return 1;
+	int i, max_sz = sizeof(FUSE_VER_NOT_IMPLEMENT);
+	const char *not_impl = FUSE_VER_NOT_IMPLEMENT;
+
+	max_sz = size < max_sz ? size : max_sz;
+
+	for (i = 0; i < max_sz; i++)
+		version[i] = not_impl[i];
+
+	version[max_sz] = '\0';
+
+	return 0;
 }
 
 struct preboot_str_size {
